axiom/optimizer/Subfields.cpp

```diff
@@ -474,19 +448,14 @@ void ToGraph::markControl(const lp::LogicalPlanNode& node) {
   }
 }
 
-void ToGraph::markAllSubfields(
-    const RowType* type,
-    const lp::LogicalPlanNode& node) {
+void ToGraph::markAllSubfields(const lp::LogicalPlanNode& node) {
   markControl(node);
 
   LogicalContextSource source = {.planNode = &node};
 
-  std::vector<const RowType*> context;
-  std::vector<LogicalContextSource> sources;
-  for (auto i = 0; i < type->size(); ++i) {
-    std::vector<Step> steps;
-    markFieldAccessed(
-        source, i, steps, /* isControl */ false, context, sources);
+  for (auto i = 0; i < node.outputType()->size(); ++i) {
+    std::vector<Step> empty;
+    markFieldAccessed(source, i, empty, /* isControl */ false, {});
```

Comment:
```suggestion
  std::vector<Step> steps;
  for (size_t i = 0; i < node.outputType()->size(); ++i) {
    markFieldAccessed(source, i, empty, /* isControl */ false, {});
    VELOX_DCHECK(steps.empty());
```
In such case steps allocation will be reused between different ordinals